source/code/scxcorelib/pal/scxfile.cpp

```diff
@@ -611,7 +611,7 @@ namespace SCXCoreLib {
         free(fp);
         fp = 0;
 #else
-        pattern = SCXFileSystem::DecodePath("/tmp/");
+        pattern = SCXFileSystem::DecodePath(tmpDir);
```

Comment:
You need to verify that tmpDir is a valid directory using SCXFile() functions. You need to validate parameters, don't assume they are correct!